api/websocket/messagebuffer/messagebuffer_test.go

```diff
@@ -0,0 +1,173 @@
+package messagebuffer
+
+import (
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/nknorg/nkn/v2/config"
+
+	"github.com/nknorg/nkn/v2/pb"
+	"github.com/stretchr/testify/require"
+)
+
+var clientId [][]byte = [][]byte{[]byte{byte(0x01)}, []byte{byte(0x02)}, []byte{byte(0x03)}, []byte{byte(0x04)}}
+
+var msg *pb.Relay = &pb.Relay{SrcIdentifier: "from", DestId: []byte("to"), Payload: []byte("payload"),
+	MaxHoldingSeconds: 10}
+var nodeSize int
+
+var msgBuffer *MessageBuffer
+
+// to create random MaxHoldingSeconds
+func addMsg(clientId []byte, m *pb.Relay) {
+	createMsgBuffer()
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+	m.MaxHoldingSeconds = uint32(r.Intn(20))
```

Comment:
We should always create a new msg and add it. We store the pointer to message, modifying it will also change previously added message

Reply:
yes. modified.

Reply:
yes. made it as a local variant.